src/NATS.Client.JetStream/Internal/NatsJSOrderedPushConsumer.cs

```diff
@@ -142,12 +142,13 @@ public async ValueTask DisposeAsync()
         _nats.ConnectionDisconnected -= OnDisconnected;
 
         _consumerCreateChannel.Writer.TryComplete();
-        _commandChannel.Writer.TryComplete();
-        _msgChannel.Writer.TryComplete();
-
         await _consumerCreateTask;
+
+        _commandChannel.Writer.TryComplete();
         await _commandTask;
 
+        _msgChannel.Writer.TryComplete();
+
```

Comment:
why did we change these? did you encounter an issue? (you're probably right to make these changes, I just can't remember my reasoning here so just curious)

Reply:
Yes, I encountered some problem.  
Before exiting WatchAsync, if Enumerable is empty, call `pushConsumer.Init();`
The consumer is initialized and the same consumer is immediately deleted in the dispose method.  
The creation operation `_consumerCreateTask` has not yet had time to complete and it requires an active `_commandChannel`, but it is already closed. Based on this, I decided to change the order of stopping channel reading.

Reply:
Otherwise, when Dispose we get `NatsJSApiException: consumer not found`.

Reply:
awesome, thanks fixing that 💯 should we stick this as comment on top of the relevant lines do you think?